jquery.overlay.js

```diff
@@ -206,63 +208,90 @@
       },
 
       renderTextOnOverlay: function () {
-        var text, i, l, strategy, match, style;
+        var text, i, l, strategy, matches, style, textContent, html, prevIndex, indexStringPairs;
         text = $('<div></div>').text(this.$textarea.val());
 
+        // Helper method. Given matches and text to match against, returns all substrings that match, and an
+        // index in the original text where each of those substrings begins.
+        function matchIndexStringPairs(matches, text) {
+          var indexStringPairs;
+          indexStringPairs = [{index: 0, string: text}];
+          matches.forEach(function(match) {
+            var nextIndexStringPairs;
+            nextIndexStringPairs = [];
+            indexStringPairs.forEach(function(pair) {
+              var matchedGroups, matchedString;
+              while ((matchedGroups = match.exec(pair.string)) !== null) {
+                matchedString = matchedGroups[0];
+                nextIndexStringPairs.push({index: pair.index + match.lastIndex - matchedString.length, string: matchedString});
+              }
+            });
+            indexStringPairs = nextIndexStringPairs;
+          });
+          return indexStringPairs;
+        }
+
         // Apply all strategies
         for (i = 0, l = this.strategies.length; i < l; i++) {
           strategy = this.strategies[i];
-          match = strategy.match;
-          if ($.isArray(match)) {
-            match = $.map(match, function (str) {
-              return str.replace(/(\(|\)|\|)/g, '\\$1');
+
+          // matching strategies are coerced to an array of one or more RegExps
+          matches = strategy.match;
+          if (!this.allowMultiPartMatching && $.isArray(matches)) {
+            matches = $.map(matches, function (str) {
+              return str.replace(/(\(|\)|\|)/g, '\$1');
+            });
+            matches = new RegExp('(' + matches.join('|') + ')', 'g');
+          }
+          if (!$.isArray(matches)) {
+            matches = [matches];
+          }
+          if (this.allowMultiPartMatching) {
+            matches = matches.map(function(match) {
+              if ($.type(match) === "string") {
+                return new RegExp(match, 'g');
+              } else {
+                return match;
+              }
             });
-            match = new RegExp('(' + match.join('|') + ')', 'g');
           }
 
           // Style attribute's string
-          style = 'background-color:' + strategy.css['background-color'];
+          style = Object.keys(strategy.css).map(function(propCSS) {
```

Comment:
nit might be clearer to use `$.each`here to iterate through the hash and push the strings to an arr instead of using `Object.keys` and `map`?